source/common/upstream/eds.cc

```diff
@@ -115,6 +115,9 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources, const std::
     const HostVector empty_hosts;
     LocalityWeightsMap empty_locality_map;
 
+    if (locality_weights_map_.size() <= i) {
```

Comment:
This was necessary in order to prevent a `EXC_BAD_ACCESS` inside `updateHostsPerLocality`. Not clear to me why this was necessary, as I'd imagine that the `locality_weights_map_` would have been large enough to hold all the priorities from previous config updates.